scripts/source-maintenance.sh

```diff
@@ -118,12 +118,12 @@ for FILENAME in `git ls-files`; do
 	#
 	# If a file includes openssl headers, then it must include compat/openssl.h
 	#
-	if test "${FILENAME}" != "compat/openssl.h" && \
-	   (FI=`grep "#include.*openssl/" "${FILENAME}" 2>/dev/null`; \
-	    test "x${FI}" != "x") && \
-	   (FI=`grep "#include \"compat/openssl\.h\"" "${FILENAME}" 2>/dev/null`; \
-	    test "x${FI}" == "x"); then
-		echo "ERROR: ${FILENAME} includes openssl headers without including \"compat/openssl.h\""
+	if test "${FILENAME}" != "compat/openssl.h"; then
+		FA=`grep "#include.*openssl/" "${FILENAME}" 2>/dev/null | head -1`;
+		FB=`grep "#include.*compat/openssl" "${FILENAME}" 2>/dev/null | head -1`;
```

Comment:
```suggestion
		FB=`grep '#include.*compat/openssl[.]h' "${FILENAME}" 2>/dev/null | head -1`;
```

... to better match the blob stated goal, the ERROR text and the intent. If we do want to relax this rule to allow for other compat/openssl headers, then we should adjust the blob stated goal and the ERROR text.

Reply:
This may not be portable grep syntax, but okay lets give it a try.